src/client/Microsoft.Identity.Client/Internal/Requests/SilentClientAuthStretegy.cs

```diff
@@ -0,0 +1,299 @@
+            }
+            catch (MsalServiceException e)
+            {
+                //Remove the account from cache in case of bad_token sub error
+                if (MsalError.BadToken.Equals(e.SubError, StringComparison.OrdinalIgnoreCase))
+                {
+                    await CacheManager.TokenCacheInternal.RemoveAccountAsync(AuthenticationRequestParameters.Account, AuthenticationRequestParameters.RequestContext).ConfigureAwait(false);
+                    logger.Warning("Failed to refresh access token because the refresh token is invalid, removing account from cache.");
+                    throw;
+                }
+
+                bool isAadUnavailable = e.IsAadUnavailable();
+
+                logger.Warning($"Refreshing the RT failed. Is AAD down? {isAadUnavailable}. Is there an AT in the cache that is usable? {cachedAccessTokenItem != null}");
+
+                if (cachedAccessTokenItem != null && isAadUnavailable)
+                {
+                    logger.Info("Returning existing access token. It is not expired, but should be refreshed.");
+                    return await CreateAuthenticationResultAsync(cachedAccessTokenItem).ConfigureAwait(false);
+                }
+
+                logger.Warning("Failed to refresh the RT and cannot use existing AT (expired or missing).");
+                throw;
+            }
+        }
+
+        private async Task<AuthenticationResult> RefreshRtOrFailAsync(CancellationToken cancellationToken)
+        {
+            // Try FOCI first
+            MsalTokenResponse msalTokenResponse = await TryGetTokenUsingFociAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            // Normal, non-FOCI flow
+            if (msalTokenResponse == null)
+            {
+                // Look for a refresh token
+                MsalRefreshTokenCacheItem appRefreshToken = await FindRefreshTokenOrFailAsync()
+                    .ConfigureAwait(false);
+
+                msalTokenResponse = await RefreshAccessTokenAsync(appRefreshToken, cancellationToken)
+                    .ConfigureAwait(false);
+            }
+            return await _silentRequest.CacheTokenResponseAndCreateAuthenticationResultAsync(msalTokenResponse).ConfigureAwait(false);
+        }
+
+        private async Task<AuthenticationResult> CreateAuthenticationResultAsync(MsalAccessTokenCacheItem cachedAccessTokenItem)
+        {
+            var msalIdTokenItem = await CacheManager.GetIdTokenCacheItemAsync(cachedAccessTokenItem.GetIdTokenItemKey()).ConfigureAwait(false);
+            return new AuthenticationResult(
+                cachedAccessTokenItem,
+                msalIdTokenItem,
+                AuthenticationRequestParameters.AuthenticationScheme,
+                AuthenticationRequestParameters.RequestContext.CorrelationId);
+        }
+
+        private void ThrowIfNoScopesOnB2C()
+        {
+            // B2C will not issue an access token if no scopes are requested
```

Comment:
I think the comment is misleading...maybe something like:
// During AT Silent with no scopes, unlike AAD, B2C will not issue an access token
...then the rest of the comment.